### PR TITLE
Update version and changelog for v1.4.0 release

### PR DESCRIPTION
**Pull request by duckontheweb**
**Description:**

Updates CHANGELOG and package version for the v1.4.0 release.

**PR Checklist:**

- [x] Code is formatted (run `pre-commit run --all-files`)
- [x] Tests pass (run `scripts/test`)
- [x] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

**Review by gadomski — APPROVED** [review]
Changlog has whitespace missing on https://github.com/stac-utils/pystac/blame/main/CHANGELOG.md#L24 (right after "owner") if you want to sidecar that change in this PR.

**Comment by duckontheweb** [comment]
I'd like to wait on this release until #704 has been merged since there may be downstream changes to `pystac-client` to fully address the issue solved by that PR.

**Pull request merged**
